File: chor-extraction/src/main/java/ast/cc/nodes/Termination.java
package ast.cc.nodes;

import ast.cc.CCVisitor;
import ast.cc.interfaces.Choreography;
import epp.MergingException;

public class Termination implements Choreography {
    private final String termination = "stop";

    @Override
    public <T> T accept(CCVisitor<T> visitor) throws MergingException {
        return visitor.visit(this);
    }

    public String getTermination() {
        return termination;
    }

    public String toString(){
        return termination;
    }
}

File: chor-extraction/src/main/java/ast/cc/nodes/Termination.kt
package ast.cc.nodes

import ast.cc.CCVisitor
import ast.cc.interfaces.Choreography
import epp.MergingException

class Termination : Choreography {
    val termination = "stop"

    @Throws(MergingException::class)
    override fun <T> accept(visitor: CCVisitor<T>): T {
        return visitor.visit(this)
    }

    override fun toString(): String {
        return termination
    }
}
